src/manimShell.ts

```diff
@@ -657,6 +662,14 @@ export class ManimShell {
                         this.eventEmitter.emit(ManimShellEvent.IPYTHON_CELL_FINISHED);
                     }
 
+                    if (this.isExecutingCommand && data.match(IPYTHON_MULTILINE_START_REGEX)) {
+                        Logger.debug(`💨 IPython multiline detected, sending extra newline`);
+                        // use sendText instead of ManimShell.exec as
+                        // shell integration does not work here
+                        // \x7F deletes the extra line ("...:") from IPython
+                        this.activeShell.sendText("\x7F");
```

Comment:
The [`ManimShell.exec()`](https://github.com/RickLuiken/manim-notebook/blob/4d31051d8ca9e7fbeb77afed3a4a9372ad27e409/src/manimShell.ts#L492-L506) method is also setting the `this.detectShellExecutionEnd` flag. Maybe add a default param to `exec()` like `forceSendTextFallback=False` (or similar, maybe you can come up with a better name) and only set it when calling it here upon a "continued IPython expression detection" (`...:`).

---

> shell integration does not work here

Also I assume you have tried it out with normal shell integration? What didn't work exactly in this case? Like even sending `\x7F` didn't get rid of the extra line?

Reply:
I did some more testing, and seems it works now with shell integration! Before when I used it, the command was not being executed, the only thing that happened was that the screen was cleared and a bunch of newlines were sent. I later added `this.isExecutingCommand` to the if statement to stop the extra newlines, and I assume that this has also fixed the issue I had with the shell integration.